src/custom/components/AffiliateStatusCheck/index.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useCallback, useEffect, useState } from 'react'
+import { useHistory } from 'react-router-dom'
+import { useActiveWeb3React } from 'hooks/web3'
+import NotificationBanner from 'components/NotificationBanner'
+import { useReferralAddress, useResetReferralAddress, useUploadReferralDocAndSetDataHash } from 'state/affiliate/hooks'
+import { useAppDispatch } from 'state/hooks'
+import { hasTrades } from 'utils/trade'
+import { SupportedChainId } from 'constants/chains'
+
+type AffiliateStatus = 'NOT_CONNECTED' | 'OWN_LINK' | 'ALREADY_TRADED' | 'ACTIVE' | 'UNSUPPORTED_NETWORK'
+
+const STATUS_TO_MESSAGE_MAPPING: Record<AffiliateStatus, string> = {
+  NOT_CONNECTED: 'Please connect your wallet to participate',
+  OWN_LINK: 'You followed your own referral link',
```

Comment:
nit: I would phrase it as:
"Your affiliate link works and It's ready to be shared!"

I find it has a more positive note (just not to feel they did sth wrong by following), and reassures the user.